step-plans/step-plans-core/src/main/java/step/core/artefacts/reports/resolvedplan/ResolvedPlanBuilder.java

```diff
@@ -37,6 +44,7 @@ private ResolvedPlanNode buildTreeRecursively(String parentId, AbstractArtefact
         artefactClone.setChildren(null);
         ResolvedPlanNode resolvedPlanNode = new ResolvedPlanNode(artefactClone, artefactHash, parentId);
         resolvedPlanNodeAccessor.save(resolvedPlanNode);
+        resolvedPlanCache.put(resolvedPlanNode.artefactHash, resolvedPlanNode.getId().toHexString());
```

Comment:
To avoid high memory consumption, we should ensure that we cannot have situations we new hashes are generated for each node during long running executions with loops

Reply:
As discussed we add here in the cache when building the resolved plan before execution. In the below method checkAndAddMissingResolvedPlanNode we exclude the work artefacts, the hash is generated in a correct way that should not generate unexpectedly different hash, so we should be safe